cognitedata/reveal
# feat: 360 annotation intersection method on viewer

#### Type of change

![Feat](https://img.shields.io/badge/Type-Feat-green)  

#### Jira ticket :blue_book:


https://cognitedata.atlassian.net/browse/REV-933

## Description :pencil:


Abandon the callback-based 360 annotation click handling in favor of the general `viewer.on('click')` events, supplimented by a new intersection method on the viewer.

Note: This breaks the API by removing the old hover- and click-event handlers from image360 collections. However, these capabilities were added in commits https://github.com/cognitedata/reveal/commit/04277d659b26238328cf203e9191687966358057#diff-85191e501e024c84e76f61c237aa2f8a3f6f0aae0bfc7c4e0d23dcf4b415578f and https://github.com/cognitedata/reveal/commit/4bcfbc8c79a5f590ef26ad3cf9964f60d5cf633f#diff-85191e501e024c84e76f61c237aa2f8a3f6f0aae0bfc7c4e0d23dcf4b415578f, which are not yet publicly available

## How has this been tested? :mag:


- Run examples
- Navigate to e.g. https://localhost:3000/?project=3d-test&env=greenfield-3d&modelId=4923307033982497&revisionId=3100284276970593
- Load `helideck-site-2-jpeg`
- Open lower-most image
- Clickyclick and see how the color of the clicked annotations changes

## Test instructions :information_source:




## Checklist :ballot_box_with_check:
